src/jit/codegenxarch.cpp

```diff
@@ -4247,241 +4247,133 @@ instruction CodeGen::genGetInsForOper(genTreeOps oper, var_types type)
-                    break;
-                case INS_ror:
-                    ins = INS_ror_N;
-                    break;
-                default:
-                    // leave 'ins' unchanged
-                    break;
-                }
-                if (isClsVarAddr)
-                {
-                    getEmitter()->emitIns_C_I(ins, attr, operand->gtClsVar.gtClsVarHnd, 0, shiftByValue);
-                } 
-                else if (isLclVarAddr)
-                {
-                    getEmitter()->emitIns_S(ins, attr, operand->gtLclVarCommon.gtLclNum, 0);
-                }
-                else if (isCnsIntOrIAndFitsWithinAddrBase)
-                {
-                    getEmitter()->emitIns_I_AI(ins, attr, shiftByValue, disp);
-                }
-                else
-                {
-                    getEmitter()->emitIns_I_AR(ins, attr, shiftByValue, operandReg, offset);
-                }
-            }
-
+            getEmitter()->emitInsRMW(ins, attr, storeInd, shiftBy);
         }
     }
     else
     {
-        // We must have the number of bits to shift
-        // stored in ECX, since we constrained this node to
-        // sit in ECX, in case this didn't happen, LSRA expects
-        // the code generator to move it since it's a single
+        // We must have the number of bits to shift stored in ECX, since we constrained this node to
+        // sit in ECX. In case this didn't happen, LSRA expects the code generator to move it since it's a single
         // register destination requirement.
         regNumber shiftReg = shiftBy->gtRegNum;
         if (shiftReg != REG_RCX)
         {
             // Issue the mov to RCX:
             inst_RV_RV(INS_mov, REG_RCX, shiftReg, shiftBy->TypeGet());
-            shiftReg = REG_RCX;
         }
 
-        // The operand to be shifted must not be in ECX
-        noway_assert(operandReg != REG_RCX);
-
-        if (isRMW)
-        {
-            if (isClsVarAddr)
-            {
-                getEmitter()->emitIns_C_R(ins, attr, operand->gtClsVar.gtClsVarHnd, shiftReg, 0);
-            }
-            else if (isLclVarAddr)
-            {
-                getEmitter()->emitIns_S_R(ins, attr, shiftReg, operand->gtLclVarCommon.gtLclNum, 0);
-            }
-            else if (isCnsIntOrIAndFitsWithinAddrBase)
-            {
-                getEmitter()->emitIns_AI_R(ins, attr, shiftReg, disp);
-            }
-            else
-            {
-                getEmitter()->emitIns_AR_R(ins, attr, indexReg, operandReg, (int) offset);
-            }
-        }
-        else
-        {
-            if (parent->gtRegNum != operandReg)
-            {
-                inst_RV_RV(INS_mov, parent->gtRegNum, operandReg, targetType);
-            }
-            inst_RV_CL(ins, parent->gtRegNum, targetType);
-        }
+        // The shiftBy operand is implicit, so call the unary version of emitInsRMW.
+        getEmitter()->emitInsRMW(ins, attr, storeInd);
```

Comment:
Yes. No actual opcode changes on x64, just a few encoding changes for shift-by-1.